lib/api/src/grpc/qdrant.rs

```diff
@@ -3475,6 +3478,13 @@ pub struct GeoRadius {
 }
 #[allow(clippy::derive_partial_eq_without_eq)]
 #[derive(Clone, PartialEq, ::prost::Message)]
+pub struct GeoPolygon {
+    /// Ordered list of coordinates representing the vertices of a polygon.
+    #[prost(message, repeated, tag = "1")]
+    pub points: ::prost::alloc::vec::Vec<GeoPoint>,
```

Comment:
Can we add validation for a minimum number of points, such as 4?

It is a bit ugly, but to do this we can likely add `length(min = 4)` here:

https://github.com/qdrant/qdrant/blob/b310eae79116a456219c81a2b2132185d78170c9/lib/api/build.rs#L78-L208

Reply:
Thanks for pointing out. Maybe we can have more discussion about the proto specification here.
From GeoJson (https://stevage.github.io/geojson-spec/#section-3.1.6), there are two validation requirements we should consider. Firstly, the minimum length of the coordinates should be 4. Secondly, the first and last coordinates should be identical.
However, it's worth noting that the Geo crate we currently rely on internally will automatically close the polygon if the last coordinate differs from the first. In order to align with the GeoJson specification and enforce a stricter requirement in the proto definition, I believe it would be beneficial to adhere to the GeoJson spec. To address this, I have made a recent update that includes a validator to ensure compliance with the GeoJson specification.
But I also would love to hear your thoughts.

Reply:
Fantastic! Great approach, better than what I suggested.

I do have a comment on where the message is placed, but I'll address that in a new review comment.

Thanks!